Applications/DataExplorer/DataView/MshView.cpp

```diff
@@ -187,8 +188,17 @@ void MshView::extractSurfaceMesh()
 		return;
 
 	const MeshLib::Mesh* mesh = static_cast<MshModel*>(this->model())->getMesh(index);
-	const MathLib::Vector3 dir(0, 0, -1);
-	static_cast<MshModel*>(this->model())->addMesh( MeshLib::MeshSurfaceExtraction::getMeshSurface(*mesh, dir, 89) );
+	SurfaceExtractionDialog dlg;
+	if (dlg.exec() != QDialog::Accepted)
+		return;
+	
+	MathLib::Vector3 const& dir (dlg.getNormal());
+	int const tolerance (dlg.getTolerance());
```

Comment:
The QSpinBox providing this value is configured to only return values in [0,90] and the extraction method want a double for calculation, so the int-value should be fine.